keps/sig-api-machinery/4006-transition-spdy-to-websockets/README.md

```diff
@@ -926,15 +1078,30 @@ Describe the metrics themselves and the reasons why they weren't added (e.g., co
 implementation difficulties, etc.).
 -->
 
-- TBD (complete for beta)
+- Metric: num_ws_remote_command_v5_upgrades_requested
+  Counts the number of times a client attempts a WebSocket/V5 RemoteCommand connection upgrade.
+  This metric would not be useful enough to justify the performance cost.
+
+- Metric: num_ws_remote_command_v5_upgrades_failed
+  Counts the number of times a client attempted a WebSockets/V5 RemoteCommand connection
+  upgrade that failed. This metric would not be useful enough to justify the performance cost.
+
+- Metric: num_ws_port_forward_v2_upgrades_requested
+  Counts the number of times a client attempts a WebSocket/V2 PortForward connection upgrade.
+  This metric would not be useful enough to justify the performance cost.
+
+- Metric: num_ws_port_forward_v2_upgrades_failed
+  Counts the number of times a client attempted a WebSockets/V2 PortForward connection
+  upgrade that failed. This metric would not be useful enough to justify the performance cost.
```

Comment:
Let's somehow also track successful ws connections.  Maybe change "num_ws_remote_command_v5_upgrades_failed" to "num_ws_remote_command_v5_upgrades" and make failed/connected a label? (same for port_forward)

Reply:
I've added four metrics for the successes and failures for websocket upgrade connections. Please let me know if you think this addresses the issue.

Reply:
Thanks. It's not clear to me if success/failure counters should be declared as separate metrics or with a single metric that has a label.   I don't see any [stable metrics](https://kubernetes.io/docs/reference/instrumentation/metrics/) where there are multiple metrics for different results, but I do see some with `result` labels.  I don't think we should block on KEP merge for this (since we're focusing on what the instrumentation should provide, not how, but we should be willing to adjust the exact metric definition during implementation)

@logicalhan Is there guidance for this?

Reply:
There is some guidance here: https://github.com/kubernetes/enhancements/pull/4456/files#r1480546416

Let's update this to match.

Reply:
Done. I've updated metrics to `...totals` with `type` containing enums `success` and `failure`. Please let me know if this addresses your feedback. Thanks.